src/components/AppModal.vue

```diff
@@ -0,0 +1,42 @@
+<template>
+  <div class="appModal">
+    <transition name="custom-classes-transition">
+      <div v-if="visible" class="appModal__content" @click.self="closeModal">
+        <div class="appModal__innerContent">
+          <component :is="component" />
+        </div>
+      </div>
+    </transition>
+  </div>
+</template>
+<script>
+import Vue from 'vue'
+import { mapState, mapMutations } from 'vuex'
+
+export default {
+  name: 'AppModal',
+  data () {
+    return {
+      component: null
+    }
+  },
+  computed: {
+    ...mapState({
+      visible: 'modalVisible',
+      modalComponent: 'modalComponent'
+    })
+  },
+  watch: {
+    modalComponent (componentName) {
+      if (!componentName) return
+      Vue.component(componentName, () => import('./modal/ModalAccount'))
```

Comment:
This will register globally the same component but with different names.
As I see, <component /> also accepts an actual component, not just it's name: https://vuejs.org/v2/api/#component.
Is not this too complicated in general? I think is cleaner to build ModalAccount as a composition of Modal component and some specific content,  like this one: https://github.com/aeternity/aepp-identity/blob/develop/src/components/AlertModal.vue